in-progress/8025-publishing-tx-objects/design.md

```diff
@@ -251,19 +252,64 @@ def submit_next_epoch_proof(proof, archive: bytes32, fees: FeePayment[EPOCH_LENG
 - TODO:
   - What are we going to do with stuff like the outhashes? They depend on the execution, so we cannot really do it at the time of the proposal if that is. So we would practically need it to happen with the entire epoch at this point. Might just be another tree to make it cheaper. It gets kinda funky with finding the block number because it depends.
 
+### Changes to circuits
+
+#### Blob circuits
+
+We need some blob circuits that allow us to prove that things were in blobs such that we can convince the contracts that i) the data was published and ii) that the data published is what have been applied to the state. Think this is likely already described somewhere else 🤷.
```

Comment:
Can you elaborate on this process? How do we convince the contracts of those two things? I see [this](https://notes.ethereum.org/@vbuterin/proto_danksharding_faq#Moderate-approach-works-with-any-ZK-SNARK) but translating it to our system would be really helpful.